allennlp/interpret/attackers/hotflip.py

```diff
@@ -0,0 +1,203 @@
+
+                # Get model predictions on new_instances, and then label the instances
+                grads, outputs = self.predictor.get_gradients(new_instances) # predictions
+                for key in outputs:
+                    if isinstance(outputs[key], torch.Tensor):
+                        outputs[key] = outputs[key].detach().cpu().numpy().squeeze().squeeze()
+                    elif isinstance(outputs[key], list):
+                        outputs[key] = outputs[key][0]
+                # add labels to new_instances
+                self.predictor.predictions_to_labeled_instances(new_instances[0], outputs)
+
+                # if the prediction has changed, then stop
+                label_change = False
+                for field in fields_to_compare:
+                    if field in new_instances[0].fields:
+
+                        equal = new_instances[0][field].__eq__(fields_to_compare[field])
+                    else:
+                        equal = outputs[field] == fields_to_compare[field]
+                    if not equal:
+                        label_change = True
+                        break
+                # if the prediction has changed, we want to return the new answer
+                # for visualization in the demo.
+                if label_change:
+                    new_prediction = get_new_prediction(outputs)
+                    break
+
+            final_tokens.append(current_tokens)
+        return \
+            sanitize({"final": final_tokens, "original": original_tokens, "new_prediction": new_prediction})
+
+# Get the model's new prediction given its outputs
+def get_new_prediction(outputs):
+    new_prediction = None
+    if "probs" in outputs: # sentiment analysis
+        new_prediction = outputs["probs"]
+    elif "label_probs" in outputs: # textual entailment
+        new_prediction = outputs["label_probs"]
+    elif "best_span_str" in outputs: # bidaf
+        new_prediction = outputs["best_span_str"]
+    elif "answer" in outputs: # NAQANet
+        ans_type = outputs["answer"]["answer_type"]
+        if ans_type == "count":
+            new_prediction = outputs["answer"]["count"]
+        else:
+            new_prediction = outputs["answer"]["value"]
+    return new_prediction
+
+def first_order_taylor(grad: numpy.ndarray,
+                       embedding_matrix: torch.nn.parameter.Parameter,
+                       token_idx: int):
+    """
+    the below code is based on
+    https://github.com/pmichel31415/translate/blob/paul/pytorch_translate/
+    research/adversarial/adversaries/brute_force_adversary.py
+
+    Replaces the current token_idx with another token_idx to increase the loss. In particular,
+    this function uses the grad, alongside the embedding_matrix to select
+    the token that maximizes the first-order taylor approximation of the loss.
+    """
+    grad = torch.from_numpy(grad)
+    embedding_matrix = embedding_matrix.cpu()
+    word_embeds = \
```

Comment:
In general, prefer splitting inside function arguments, or using parens, instead of adding a backslash.  You have this in a bunch of places.